world_boss/app/kms.py

```diff
@@ -210,17 +210,20 @@ def query_balance(self, headless_url: str, currency: CurrencyDictionary) -> str:
                     ds.StandaloneQuery.stateQuery.select(
                         ds.StateQuery.balance.args(
                             address=self.address,
-                            currency={
-                                "ticker": "CRYSTAL",
-                                "decimalPlaces": 18,
-                            },
-                        ).select(ds.FungibleAssetValueWithCurrencyType.quantity)
+                            currency=currency,
+                        ).select(
+                            ds.FungibleAssetValueWithCurrencyType.quantity,
+                            ds.FungibleAssetValueWithCurrencyType.currency.select(
+                                ds.CurrencyType.ticker
+                            ),
+                        )
                     )
                 )
             )
             result = session.execute(query)
             balance = result["stateQuery"]["balance"]["quantity"]
-            return f'{balance} {currency["ticker"]}'
+            ticker = result["stateQuery"]["balance"]["currency"]["ticker"]
+            return f"{balance} {ticker}"
```

Comment:
(Even in future PR) it would be better if we handle this value as a well-formed struct instead of `str` and defer to render(convert to `str`) as late as possible.